### PR TITLE
EVG-2713: move host termination into amboy job

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: units/host_termination.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+	idleTimeStartsAt := j.host.LastTaskCompletedTime
+	if idleTimeStartsAt.IsZero() || idleTimeStartsAt == util.ZeroTime {
+		idleTimeStartsAt = j.host.StartTime
+	}
+
+	// clear the running task of the host in case one has been assigned.
+	if j.host.RunningTask != "" {
+		grip.Warning(message.Fields{
+			"runner":   "monitor",
```
Comment (john-m-liu) [inline units/host_termination.go]: Same here: should we log job type rather than monitor?

File: units/stats_host_idle.go [hunk 2]
```diff
@@ -51,6 +53,17 @@ func newHostIdleJob() *collectHostIdleDataJob {
 	return j
 }
 
+func newHostIdleJobForTermination(env evergreen.Evergreen, settings *evergreen.Settings, manager cloud.CloudManager,
```
Comment (john-m-liu) [inline units/stats_host_idle.go]: evergreen.Environment

File: units/stats_host_idle.go [hunk 3]
```diff
@@ -95,26 +108,31 @@ func (j *collectHostIdleDataJob) Run(ctx context.Context) {
 		j.env = evergreen.GetEnvironment()
 	}
 
-	settings := j.env.Settings()
+	if j.settings == nil {
```
Comment (john-m-liu) [inline units/stats_host_idle.go]: Why is this needed? The Settings() function won't load them if it's nil (and it shouldn't be nil if you got to here)

File: units/host_termination.go (hunk 1 above)
Comment (tychoish) [reply units/host_termination.go]: done

File: units/stats_host_idle.go (hunk 2 above)
Comment (tychoish) [reply units/stats_host_idle.go]: done

File: units/stats_host_idle.go (hunk 3 above)
Comment (tychoish) [reply units/stats_host_idle.go]: There are three ways this job can be constructed: 
1. Directly via the public `NewCollectHostIdleDataJob` (j.settings is nil,) 
2. Directly via the private constructor (when called directly from another job: j.settings is populated.) 
3. Via the factory in the job registry, (j.settings is nill and cannot be populated.) 

So we have to be careful to populate the attribute when we execute the job.

Comment (john-m-liu) [reply units/stats_host_idle.go]: But the function will just return nil if if settings is nil which is fine. Also this should be j.env.settings if you're going to check that

Comment (tychoish) [reply units/stats_host_idle.go]: the case I'm protecting against is the case where the job is instantiated without a settings object (which I think is fairly common) and we need to pull it from the environment. 

In one major case, the object will be instantiated with a settings object, and in those cases, we shouldn't re-requery.

**Comment by john-m-liu** [comment]
lgtm after fixing compile
